### PR TITLE
Add "Unexpected argument" message (issue #4974)

### LINKED ISSUE #4974: [Feature Request] useful error when unexpected argument
### What is the issue?

I got stuck for a long time because I had `--peer-port` instead of `--peerport` as an argument. It would be nice if transmission could print an error about an unexpected argument. Right now it just prints the help page. 

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

4.0.0

### PR DESCRIPTION
**Pull request by RoeyFuchs**

### CODE REVIEW

File: daemon/daemon.cc [hunk 1]
```diff
@@ -628,6 +628,12 @@ bool tr_daemon::parse_args(int argc, char const* const* argv, bool* dump_setting
             tr_variantDictAddBool(&settings_, TR_KEY_utp_enabled, false);
             break;
 
+        case TR_OPT_UNK:
+            fprintf(stderr, "Unexpected argument: %s \n", optstr);
+            tr_getopt_usage(MyName, Usage, std::data(Options));
+            *exit_code = 0;
```
Comment (maxz) [inline daemon/daemon.cc:634]: This value should probably be 64 instead of 0.

**Comment by maxz** [comment]
The exit code should not be 0 here but rather 64. I could not find any Transmission specific errors codes and in the case of an unexpected argument it would not make real sense to use a project specific code.
Maybe even the symbol EX_USAGE could be used but I'm not sure about the portability of that one so the number is safer.

See `/usr/include/sysexits.h` on GNU/Linux or `man sysexits` on BSD:
`#define EX_USAGE        64      /* command line usage error */`

Maybe this should also be adjusted for the default case, but @ckerr would have to give his opinion there.

File: daemon/daemon.cc (hunk 1 above)
Comment (ckerr) [reply daemon/daemon.cc:634]: why 64?

Comment (ckerr) [inline daemon/daemon.cc:634]:
```suggestion
            *exit_code = 1;
```

**Review by ckerr — CHANGES_REQUESTED** [review]

**Review by ckerr — APPROVED** [review]

**Comment by ckerr** [comment]
Thanks @RoeyFuchs!

Notes: Made the "unrecognized argument" error message more readable.

**Pull request merged**